libs/parallelism/algorithms/include/hpx/parallel/algorithms/inclusive_scan.hpp

```diff
@@ -499,32 +398,496 @@ namespace hpx { namespace parallel { inline namespace v1 {
+#endif
+    }
 
-        typedef hpx::traits::is_segmented_iterator<FwdIter1> is_segmented;
+    // clang-format off
+    template <typename ExPolicy, typename FwdIter1, typename FwdIter2,
+        typename Op,
+        HPX_CONCEPT_REQUIRES_(
+            hpx::is_execution_policy<ExPolicy>::value &&
+            hpx::traits::is_iterator<FwdIter1>::value &&
+            hpx::traits::is_iterator<FwdIter2>::value &&
+            hpx::is_invocable_v<Op,
+                typename std::iterator_traits<FwdIter1>::value_type,
+                typename std::iterator_traits<FwdIter1>::value_type
+            >
+        )>
+    // clang-format on
+    HPX_DEPRECATED_V(1, 8,
+        "hpx::parallel::inclusive_scan is deprecated, use hpx::inclusive_scan "
+        "instead")
+        typename util::detail::algorithm_result<ExPolicy, FwdIter2>::type
+        inclusive_scan(ExPolicy&& policy, FwdIter1 first, FwdIter1 last,
+            FwdIter2 dest, Op&& op)
+    {
+        static_assert((hpx::traits::is_forward_iterator<FwdIter1>::value),
+            "Requires at least forward iterator.");
+        static_assert((hpx::traits::is_forward_iterator<FwdIter2>::value),
+            "Requires at least forward iterator.");
+
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+        return hpx::parallel::v1::detail::inclusive_scan<FwdIter2>().call(
+            std::forward<ExPolicy>(policy), first, last, dest,
+            std::forward<Op>(op));
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic pop
+#endif
+    }
+
+    // clang-format off
+    template <typename ExPolicy, typename FwdIter1, typename FwdIter2,
+        HPX_CONCEPT_REQUIRES_(
+            hpx::is_execution_policy<ExPolicy>::value &&
+            hpx::traits::is_iterator<FwdIter1>::value &&
+            hpx::traits::is_iterator<FwdIter2>::value
+        )>
+    // clang-format on
+    HPX_DEPRECATED_V(1, 8,
+        "hpx::parallel::inclusive_scan is deprecated, use hpx::inclusive_scan "
+        "instead")
+        typename util::detail::algorithm_result<ExPolicy, FwdIter2>::type
+        inclusive_scan(
+            ExPolicy&& policy, FwdIter1 first, FwdIter1 last, FwdIter2 dest)
+    {
+        static_assert((hpx::traits::is_forward_iterator<FwdIter1>::value),
+            "Requires at least forward iterator.");
+        static_assert((hpx::traits::is_forward_iterator<FwdIter2>::value),
+            "Requires at least forward iterator.");
 
-        return detail::inclusive_scan_(std::forward<ExPolicy>(policy), first,
-            last, dest, std::plus<value_type>(), is_segmented(),
-            util::projection_identity{});
+        using value_type = typename std::iterator_traits<FwdIter1>::value_type;
+
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
```

Comment:
We should move this (and the corresponding `pop` below) into separate include files to reduce cluttering the code (place them here, for instance: https://github.com/STEllAR-GROUP/hpx/tree/master/libs/core/config/include/hpx/config). But that's something for a separate PR.